src/workspace-tree/bazel_workspace_tree_provider.ts

```diff
@@ -84,9 +84,11 @@ export class BazelWorkspaceTreeProvider
       // individual top level items.
       return Promise.resolve(
         vscode.workspace.workspaceFolders.map((folder) => {
-          return new BazelWorkspaceFolderTreeItem(
-            BazelWorkspaceInfo.fromWorkspaceFolder(folder),
-          );
+          const workspaceInfo = BazelWorkspaceInfo.fromWorkspaceFolder(folder);
+          if (workspaceInfo) {
+            return new BazelWorkspaceFolderTreeItem(workspaceInfo);
+          }
+          return undefined;
```

Comment:
This will put `undefined` elements in the array that the promise resolves to, which might end up being harmless but we shouldn't depend on them being ignored.

Since `flatMap` isn't well supported yet apparently (it's new in ES2019), you'll need to chain a `filter` after the `map` call:

```typescript
vscode.workspace.workspaceFolders.map((folder) => {
  ...
}).filter((folder) => { folder !== undefined }),
```

Reply:
It does get ignored but I agree that it feels a little hopeful without the filter. Added.